src/main/java/com/bieganski/ox/PositionCalculator.java

```diff
@@ -15,6 +15,11 @@ int calculatePosition(String userInput) {
         String[] coordinates = userInput.split(" ");
         String row = coordinates[0];
         String column = coordinates[1];
-        return Integer.parseInt(row) * boardSize + Integer.parseInt(column);
+        try {
+            return Integer.parseInt(row) * boardSize + Integer.parseInt(column);
+        } catch (NumberFormatException ex){
+            App.LOG.warn("User provided incorrect number");
+            return -1;
+        }
     }
 }
```

Comment:
The same as up. Think about new class to check user input instead make it inside the method.